extensions/series/src/lib.rs

```
@@ -0,0 +1,137 @@
+            args.get(1).map(|v| v.to_integer().unwrap_or(i64::MAX)),
+            ResultCode::InvalidArgs
+        );
+        let step = try_option!(
+            args.get(2).map(|v| v.to_integer().unwrap_or(1)),
+            ResultCode::InvalidArgs
+        );
+        cursor.start = start;
+        cursor.current = start;
+        cursor.step = step;
+        cursor.stop = stop;
+        ResultCode::OK
+    }
+
+    fn column(cursor: &Self::VCursor, idx: u32) -> Value {
+        cursor.column(idx)
+    }
+
+    fn next(cursor: &mut Self::VCursor) -> ResultCode {
+        GenerateSeriesCursor::next(cursor)
+    }
+
+    fn eof(cursor: &Self::VCursor) -> bool {
+        cursor.eof()
+    }
+}
+
+/// The cursor for iterating over the generated sequence
+#[derive(Debug)]
+struct GenerateSeriesCursor {
+    start: i64,
+    stop: i64,
+    step: i64,
+    current: i64,
+    error: Option<ResultCode>,
+}
+
+impl GenerateSeriesCursor {
+    fn next(&mut self) -> ResultCode {
+        let current = self.current;
+
+        // Check if we've reached the end
+        if (self.step > 0 && current >= self.stop) || (self.step < 0 && current <= self.stop) {
+            return ResultCode::EOF;
+        }
+
+        self.current = current.saturating_add(self.step);
+        ResultCode::OK
+    }
+}
+
+impl VTabCursor for GenerateSeriesCursor {
+    type Error = ResultCode;
+
+    fn next(&mut self) -> ResultCode {
+        GenerateSeriesCursor::next(self)
+    }
+
+    fn eof(&self) -> bool {
+        (self.step > 0 && self.current > self.stop) || (self.step < 0 && self.current < self.stop)
+    }
+
+    fn column(&self, idx: u32) -> Value {
+        match idx {
+            0 => Value::from_integer(self.current),
+            1 => Value::from_integer(self.start),
+            2 => Value::from_integer(self.stop),
+            3 => Value::from_integer(self.step),
+            _ => Value::null(),
+        }
+    }
+
+    fn error(&self) -> Option<Self::Error> {
```

I'd like opinions on this.. This extension is perhaps doesn't have any great examples of when you would need to propagate an error, but is having an (optional) error/set_error method valuable here? When it's not enough to just return result codes?

---

I think it gives additional flexibility, without demanding a lot more from the extension developer. Would this error just be propagated, converted to a String, and be printed out to the user? Or is there some other functionality that it would serve?

---

it would be there so you could set the error at any time during an operation and it would be checked automatically by core when it throws so we can provide a better error message